src/main/java/org/dependencytrack/policy/cel/CelPolicyEngine.java

```diff
@@ -157,7 +157,7 @@ public void evaluateProject(final UUID uuid) {
 
             // Preload licenses for the entire project, as chances are high that they will be used by multiple components.
             final Map<Long, org.hyades.proto.policy.v1.License> licenseById;
-            if (requirements.containsKey(TYPE_LICENSE)) {
+            if (requirements.containsKey(TYPE_LICENSE) || requirements.containsKey(TYPE_COMPONENT)) {
```

Comment:
While adding this check resolves the issue, it will cause licenses to always be loaded, which is not necessarily what we want.

If you set a breakpoint here:

https://github.com/DependencyTrack/hyades-apiserver/blob/777c9f170c849d7071b721ec0258122b668e0708/src/main/java/org/dependencytrack/policy/cel/CelPolicyScriptHost.java#L135-L136

You'll see that we correctly identified that `resolved_license` is being accessed. But because this is an existence check (`has(component.resolved_license)`), the `License` type itself is not present in `requirements`:

<img width="642" alt="image" src="https://github.com/DependencyTrack/hyades-apiserver/assets/5693141/6cd9e47c-6753-4166-9d3d-a131769fdc29">

Instead of checking on the `Component` type, I'd suggest to check for `Component.resolved_license` being accessed:

```suggestion
            if (requirements.containsKey(TYPE_LICENSE) || (requirements.containsKey(TYPE_COMPONENT) && requirements.get(TYPE_COMPONENT).contains("resolved_license"))) {
```